### PR TITLE
Update Posit Connect version to 2024.04.0 and fix `update-rsc-versions`

### PR DESCRIPTION
**Pull request by Lytol**
- Update Posit Connect version to 2024.04.0
- Update the Justfile task for `update-rsc-versions` to remove connect/Justfile and connect-content-init/Justfile
- Also, no need to change the docker-compose.yml since it uses the RSC_VERSION var directly

### CODE REVIEW

**Review by bschwedler — APPROVED** [review]
LGTM

**Comment by dbkegley** [comment]
We'll also be upgrading the default quarto installation to `v1.4.552` as part of this release.

**Comment by bschwedler** [comment]
@dbkegley @Lytol I just changed the base branch to `dev` so we can get the workbench release out at the same time.

Can you check to make sure my rebase didn't bork anything up?

**Comment by Lytol** [comment]
> @dbkegley @Lytol I just changed the base branch to `dev` so we can get the workbench release out at the same time.
> 
> Can you check to make sure my rebase didn't bork anything up?

Looks good to me!

File: docker-bake.hcl [hunk 1]
```diff
@@ -101,7 +101,7 @@ variable PACKAGE_MANAGER_BUILD_MATRIX {
 variable CONNECT_BUILD_MATRIX {
     default = {
         builds = [
-            {os = "ubuntu2204", r_primary = "4.2.3", r_alternate = "4.1.3", py_primary = "3.9.17", py_alternate = "3.8.17", quarto = "1.3.340"},
+            {os = "ubuntu2204", r_primary = "4.2.3", r_alternate = "4.1.3", py_primary = "3.9.17", py_alternate = "3.8.17", quarto = "DEFAULT_QUARTO_VERSION"},
```
Comment (bschwedler) [inline docker-bake.hcl:104]: I _think_ we'll need to reference this as a variable, or not put it in quotes.
```suggestion
            {os = "ubuntu2204", r_primary = "4.2.3", r_alternate = "4.1.3", py_primary = "3.9.17", py_alternate = "3.8.17", quarto = "${DEFAULT_QUARTO_VERSION}"},
```

File: docker-bake.preview.hcl [hunk 2]
```diff
@@ -130,7 +130,7 @@ variable PACKAGE_MANAGER_BUILD_MATRIX {
 variable CONNECT_BUILD_MATRIX {
     default = {
         builds = [
-            {os = "ubuntu2204", r_primary = "4.2.3", r_alternate = "4.1.3", py_primary = "3.11.9", py_alternate = "3.10.14", quarto = "1.3.340"},
+            {os = "ubuntu2204", r_primary = "4.2.3", r_alternate = "4.1.3", py_primary = "3.11.9", py_alternate = "3.10.14", quarto = "DEFAULT_QUARTO_VERSION"},
```
Comment (bschwedler) [inline docker-bake.preview.hcl:133]: Same as above comment.

**Review by bschwedler — CHANGES_REQUESTED** [review]

File: docker-bake.hcl (hunk 1 above)
Comment (bschwedler) [reply docker-bake.hcl:104]: The alternative
```suggestion
            {os = "ubuntu2204", r_primary = "4.2.3", r_alternate = "4.1.3", py_primary = "3.9.17", py_alternate = "3.8.17", quarto = DEFAULT_QUARTO_VERSION},
```

**Review by bschwedler — APPROVED** [review]

**Pull request merged**